Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -3557,6 +3563,91 @@ def visit_ExprNode(self, node):
     visit_Node = VisitorTransform.recurse_to_children
 
 
+class _ThreadsafetyVisitAssignmentTargetNodes(TreeVisitor):
```

Comment:
ISTM that this whole class could be folded into a single, simple `isinstance()` check. I's non-recursive and I could only find a single usage that does "create, run, discard". We don't need the full node type dispatch mechanism for just that.